[MIYAD-6151] bug fix:

AvivgDu opened the pull request:
1. Remove assert in case there is no start_timestamp.
2. Add bad pipeline state option.
3. Set bad pipeline state in case there is no no appsink.
4. Clear _startTimestampQueue when stopPipeline is called and print queue size.

tuvalkohl commented on src/CameraGstreamer.h:
```diff
@@ -93,7 +93,7 @@ class CameraGstreamer {
     void removeBusWatch();
     static const std::unordered_map< std::string, void (CameraGstreamer::*)( const char *key, const char *value ) > optionHandlers;
-
+    void clearStartTimestampQueue();
```
Please don't mix methods with members. `optionHandlers` is a static member.

```
    void removeBusWatch();
    void clearStartTimestampQueue();

    static const std::unordered_map< std::string, void (CameraGstreamer::*)( const char *key, const char *value ) > optionHandlers;

tuvalkohl commented on src/CameraGstreamer.cpp:
```diff
@@ -172,6 +179,7 @@ void CameraGstreamer::stopPipeline() {
         gst_element_set_state( _currentPipelineElement, GST_STATE_NULL );
         gst_object_unref(_currentPipelineElement);
         removeBusWatch();
+        clearStartTimestampQueue();
```
What was the effect of not having this?
This seems like a separate bugfix, which should be in a separate commit (with an explanation in the commit message, what was the possible effect of previously not clearing the queue)

tuvalkohl commented on src/CameraGstreamer.cpp:
```diff
@@ -142,13 +142,20 @@ void CameraGstreamer::playGetVideoPackets() {
+            }
+        } else {
+            std::cout << "[ERROR] [camera " << _cameraId << "] camera appsink was not set." << std::endl;
+            _manager->badPipeline(_cameraId);
```
Don't we want to return after this line?
Does it make sense to `startPipeline()` if you know you don't have an `appsink`?
Don't we want to go to NO_SIGNAL?

tuvalkohl commented on src/CameraGstreamer.cpp:
```diff
@@ -403,3 +411,9 @@ bool CameraGstreamer::parseBoolValue( const char *key, const char *value ) {
 }
+
+void CameraGstreamer::clearStartTimestampQueue() {
+   std::cout << "[camera "<<_cameraId<<"] clear startTimestampQueue, queue size: " << _startTimestampQueue.size() << std::endl;
```
Sorry for the nitpick, but please capitalize the logs (always), i.e `clear` -> `Clear`:
```
std::cout << "[camera "<<_cameraId<<"] Clear startTimestampQueue, queue size: " << _startTimestampQueue.size() << std::endl;

tuvalkohl commented on src/CameraGstreamer.cpp:
```diff
@@ -142,13 +142,20 @@ void CameraGstreamer::playGetVideoPackets() {
+                g_signal_connect( appsink, "eos", G_CALLBACK(onEos), this );
+            }
+        } else {
+            std::cout << "[ERROR] [camera " << _cameraId << "] camera appsink was not set." << std::endl;
```
I suggest:
```
std::cout << "[ERROR] [camera " << _cameraId << "] Failed to find 'appsink' in pipeline." << std::endl;

tuvalkohl reviewed (commented):
Looks good, just:
1. Please explain why the `_startTimestampQueue` was necessary, and I think it should be in a separate commit
2.  Please make the commit message more verbose - the title should give some information, and the body should explain what you fixed:
```
[bugfix][MIYAD-6151]: Prevent Gstreamer from logging failed asserts for every pipeline without 'start_timestamp'

Until now, the following errors would be logged in the Streamer journal
for every GStreamerr pipeline which didn't contain 'start_timestamp' 
(which is supposed to be all pipelines except for one):
    gst_element_get_static_pad: assertion 'GST_IS_ELEMENT (element)' failed
    gst_pad_add_probe: assertion 'GST_IS_PAD (pad)' failed
    [camera 5] no start_timestamp name was set.

Now these errors will no longer be logged, and we will only 
log when we do find 'start_timestamp'.

Additionally:
* Report camera state 10001 if the pipeline does not contain 'appsink' (previously this error was ignored by the camera lib)
* Clear the startTimestamp queue whenever the Pipeline is stopped (TODO AVIV - explain why this is necessary, and probably should split it to an independent commit)

AvivgDu committed:
[bugfix][MIYAD-6151]: Prevent Gstreamer logging failed asserts for every pipeline without 'start_timestamp'

Until now, the following errors would be logged in the Streamer journal
for every GStreamerr pipeline which didn't contain 'start_timestamp'
(which is supposed to be all pipelines except for one):
    gst_element_get_static_pad: assertion 'GST_IS_ELEMENT (element)' failed
    gst_pad_add_probe: assertion 'GST_IS_PAD (pad)' failed
    [camera 5] no start_timestamp name was set.

Now these errors will no longer be logged, and we will only
log when we do find 'start_timestamp'.

Additionally:
* Report camera state 10001 if the pipeline does not contain 'appsink' (previously this error was ignored by the camera lib)

tuvalkohl commented:
@AvivgDu - should this PR be closed?
